src/lib.rs

```diff
@@ -153,12 +227,28 @@ pub struct Interrupt {
 }
 
 impl Interrupt {
-    fn parse(tree: &Element) -> Interrupt {
-        Interrupt {
-            name: try!(tree.get_child_text("name")),
-            description: tree.get_child_text("description"),
-            value: try!(parse::u32(try!(tree.get_child("value")))),
-        }
+    fn parse(tree: &Element) -> Result<Interrupt> {
+        let name = bail_if_none!(
+            tree.get_child_text_try("name")?,
+            "Couldn't find name of `interrupt`",
```

Comment:
Never mind all these comments. I had not looked at the produced error messages just yet.